deps/build.jl

```diff
@@ -15,8 +15,17 @@ module DefaultDeps
     end
 end
 
-ROOTENV = get(ENV, "CONDA_JL_HOME", DefaultDeps.ROOTENV)
 MINICONDA_VERSION = get(ENV, "CONDA_JL_VERSION", DefaultDeps.MINICONDA_VERSION)
+ROOTENV = get(ENV, "CONDA_JL_HOME") do
+    root = DefaultDeps.ROOTENV
+
+    # Ensure the ROOTENV uses the current MINICONDA_VERSION when not using a custom ROOTENV
+    if normpath(dirname(root)) == normpath(condadir) && all(isdigit, basename(root)) && basename(root) != MINICONDA_VERSION
```

Comment:
In the current form it's safe to remove the `basename(root) != MINICONDA_VERSION` as the result is the same:
```suggestion
    if normpath(dirname(root)) == normpath(condadir) && all(isdigit, basename(root))
```